File: src/node/Producer.js
const kafka = require('kafka-node');
const Producer = kafka.Producer;
const client = new kafka.KafkaClient('kafka:9094');
const producer = new Producer(client);

payloads = [
    { topic: 'test', messages: 'hi'}
];

producer.on('ready', function () {
    
    producer.send(payloads, function (err, data) {
        console.log(data);
    });
}).on('error', function (err) {
    console.log(err);
})

File: src/producer.py
from kafka import KafkaProducer
from kafka.errors import KafkaError

producer = KafkaProducer(bootstrap_servers=['kafka:9092'])

topic = "test"

print(producer)
# Asynchronous by default
future = producer.send(topic=topic, key=b'foo', value=b'megyez')
print(future)
try:
    a = 1
    record_metadata = future.get(timeout=60)
    # Successful result returns assigned partition and offset
    print(record_metadata.topic)
    print(record_metadata.partition)
    print(record_metadata.offset)
except KafkaError as e:
    print(e)


